pkg/network/proxy/hybridproxier.go

```diff
@@ -274,10 +268,9 @@ func (p *HybridProxier) OnEndpointsAdd(endpoints *corev1.Endpoints) {
 }
 
 func (p *HybridProxier) OnEndpointsUpdate(oldEndpoints, endpoints *corev1.Endpoints) {
-	// we track all endpoints in the unidling endpoints handler so that we can succesfully
-	// detect when a service become unidling
-	klog.V(6).Infof("hybrid proxy: (always) update ep %s/%s in unidling proxy", endpoints.Namespace, endpoints.Name)
+	klog.V(6).Infof("hybrid proxy: update ep %s/%s", endpoints.Namespace, endpoints.Name)
 	p.unidlingProxy.OnEndpointsUpdate(oldEndpoints, endpoints)
+	p.mainProxy.OnEndpointsUpdate(oldEndpoints, endpoints)
```

Comment:
any possible performance implications related to the duplication?

Reply:
The change that you are commenting on shouldn't really have noticeable performance implications. Unless you have a cluster with LOTS of idled services, the main proxy would be receiving almost all of the events anyway. It might even end up helping performance by getting rid of pointless allocations and frees...

OTOH, the "watch both Endpoints and EndpointSlices" commit does have performance implications... we'll now have both Endpoints and EndpointSlice informers, and so will be caching both sets of objects.

Perhaps if we can use Services rather than Endpoints to track idling, then we could cheat and just synthesize fake Endpoints from the EndpointSlice events to pass to the unidling proxy, rather than actually listening for real Endpoints events?

Reply:
we do have annotations on services 
https://github.com/openshift/oc/pull/720

Reply:
https://github.com/openshift/openshift-controller-manager/commit/dca955d79996d6c9e8de574a79b699d2d3437d8d

Reply:
I am strongly in favor of at least attempting to convert endpoints to synthetic EndpointSlices.  Double watching doubles memory usage and CPU and network traffic to control plane on all nodes on big dense clusters.  It might (for instance) blow up a very large cluster.